UnityServer/functions/src/model/UnityManager.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as firebase from "firebase-admin";
+import { UnityStatus } from "./UnityStatus";
+import * as serviceAccountPlay from "../keys/placeholderKey.json";
+var Verifier = require('google-play-billing-validator');
+
+const FIRESTORE_USERS_COLLECTION = 'users';
+const FIRESTORE_PURCHASES_COLLECTION = 'purchases';
+
+enum SkuType {
+  ONE_TIME = 'inapp',
+  SUBS = 'subs'
+}
+/* UnityManager is part of Model layer.
+ * It manages request of users from the unity sample game.
+ */
+export class UnityManager {
```

Comment:
I think some of the methods in this class could benefit from some comments to explain what each method is doing and how it's doing it to make it easier for developers to follow what's going on.

Reply:
Sure, will add some comments